vmdb/app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -916,6 +851,67 @@ def prov_set_form_vars(req = nil)
     end
   end
 
+  def workflow_type_from_vars(req)
+    if ["miq_template", "service_template", "vm"].include?(@edit[:org_controller])
+      if params[:prov_type] && !req  # only do this new requests
+        @edit[:prov_id] = params[:prov_id]
+        options         = {}
+        wf_type         = if params[:prov_type] == "migrate"
```

Comment:
Suggest converting this massive `if` into a method `workflow_class_from_vars` or `workflow_type_from_vars`.